lib/iex/lib/iex/helpers.ex

```diff
@@ -80,8 +80,9 @@ defmodule IEx.Helpers do
     if mix_started? do
       config = Mix.Project.config
       reenable_tasks(config)
-      apps = stop_apps(config)
-      Mix.Task.run("app.start")
+      {not_started?, apps} = stop_apps(config)
+      args = if not_started?, do: ["--no-start"], else: []
+      Mix.Task.run("app.start", args)
       {:restarted, apps}
```

Comment:
Great question! Let's return accordingly to `not_started?`. The reason why I wanted to return `:restarted` was to make it clear they were restarted, so I don't want to lose this distinction. :)